src/components/IndexTable/IndexTable.tsx

```diff
@@ -204,6 +205,27 @@ function IndexTableBase({
     [resizeTableScrollBar],
   );
 
+  const [canScrollRight, setCanScrollRight] = useState(true);
+
+  const setRightScroll = useCallback(() => {
+    if (
+      !lastColumnSticky ||
+      !tableElement.current ||
+      !scrollableContainerElement.current
+    ) {
+      return;
+    }
+
+    const tableRect = tableElement.current.getBoundingClientRect();
+    const scrollableRect = scrollableContainerElement.current.getBoundingClientRect();
+
+    setCanScrollRight(tableRect.width > scrollableRect.width);
+  }, [lastColumnSticky, setCanScrollRight]);
```

Comment:
Set state functions don't need to be included as a dependency!

```suggestion
  }, [lastColumnSticky]);
```